src/Octoshift/Services/GithubApi.cs

```diff
@@ -104,6 +104,48 @@
         await _retryPolicy.Retry(() => _client.DeleteAsync(url));
     }
 
+    public virtual async Task<string> GetLoginName()
+    {
+        var url = $"{_apiUrl}/graphql";
+
+        var payload = new
+        {
+            query = "query{viewer{login}}"
+        };
+
+        try
+        {
+            return await _retryPolicy.Retry(async () =>
+            {
+                var data = await _client.PostGraphQLAsync(url, payload);
+
+                return (string)data["data"]["viewer"]["login"];
+            });
+        }
+        catch (Exception ex)
+        {
+            throw new OctoshiftCliException($"Failed to lookup the login for current user", ex);
+        }
```

Comment:
@synthead The OctoshiftCliException is set up so that we can pass the exception ("ex") as an `inner exception`; which the user can view using the `--verbose` option. :) 

This is the pattern used on a few other functions within this class. If we want to get more specific with errors here, should this and other functions be cleaned up as part of a separate PR? 🤷‍♀️ 

As for "what does a failure" look like here; honestly, I don't think there really will be anything outside of a network error; I set up the catch to be cautious. Since we're just getting the user's login and that is associated with the PAT they are using, I don't think we should run into a permissions issue or anything else particularly funky.